### ISSUE TITLE
Add reserve a DOI functionality

### ISSUE THREAD

**Issue by dtarb** [hydroshare/hydroshare#2558]
Consider enabling a "reserve DOI" capability that would add on a resource landing page how to cite information about the resource once it has a DOI.  While HydroShare DOI's are constructed from their GUID, and this effectivley reserves the DOI already, this is opaque to users.  Reserve DOI would show users the DOI citable form of the resource under some sort of pending label, e.g. "A DOI has been reserved for this resource.  This DOI reference will not be active until this resource has been accepted and formally published".  This would inform users of the future citation for them to include in papers under review.  Then when the paper is published, they can add a reference to the resource that cites the paper and formally publish it.

**Comment by aphelionz** [comment]
Seems extremely easy to do. I will consider adding this to the next CUAHSI sprint. Although I'm sure @Maurier can knock this out in under an hour.

**Comment by aphelionz** [comment]
@dtarb Are there any actions necessary for the "reservation" of a DOI? It seems we can just construct it from the resource ID without taking any action. Once a resource is created and assigned its HydroShare guid, we know what the DOI is going to be, right?

**Comment by hyi** [comment]
@aphelionz I can confirm this is correct, DOI for a resource can be constructed from resource GUID with a fixed prefix string.

**Comment by dtarb** [comment]
Yes.  This is all about making it clear to a user what the DOI for the resource will be so they can use it in papers before they actually get the DOI registered at Crossref.

**Comment by horsburgh** [comment]
@aphelionz - I've looked at another system that has this functionality. Once a resource has been created, there is simply a link on the page that says "Reserve Digital Object Identifier (DOI)". When that is clicked, the DOI is displayed with a note that says that the DOI will become active when the item is formally published.

I think the "Reserve Digital Object Identifier (DOI)" link should show up just under the "How to Cite" section on the resource landing page. Once reserved, I think there should be a note under the citation that states something like: "You have reserved a DOI for your resource. Once your resource has been formally published, the DOI will become active and the citation link for your resource will be: http://dx.doi.org/10.4211/hs.XXXXXXXXXXXXXXXX."

**Comment by aphelionz** [comment]
We are not really reserving anything, are we though?

**Comment by horsburgh** [comment]
Interesting question. Technically, the step of "reserving the DOI" is done when the resource is created because the DOI is based on the GUID assigned to the resource. No other resource will ever have the same GUID, so the DOI is effectively reserved. 

So, I suppose we could dispense with the "Reserve DOI" step entirely and just add a note below the How to Cite section for unpublished resources that says something like:

"Once your resource has been formally published, a digital object identifier (DOI) will become active, and the citation link for your resource will be: http://dx.doi.org/10.4211/hs.XXXXXXXXXXXXXXXX. When your resource's metadata and content files are complete, you can formally publish the resource by clicking on the Publish button at the top of the page." 

This message should only be displayed for resource owners.

**Comment by hyi** [comment]
@dtarb I like the suggestion by @horsburgh since once a resource is created, the DOI is actually "reserved" by default based on our DOI assignment rule in HydroShare since this DOI will never be taken by others. The main issue is to inform the users which the DOI for their resource would be once they take the steps to publish the resource.

**Comment by dtarb** [comment]
I agree with @horsburgh suggestion.  One technical nicety though, the link to the DOI should not be constructed as a hyperlink because until the DOI is registered it will be a broken link.  So the blue hypertext link in the quote above should just be plain text.

**Comment by aphelionz** [comment]
@dtarb roger that

**Comment by aphelionz** [comment]
@dtarb @horsburgh  like this?

![image](https://user-images.githubusercontent.com/106148/35107922-36275072-fc40-11e7-9bce-0c0d2e42e6bd.png)

**Comment by dtarb** [comment]
Looks good.  I suggest spell out "digital object identifier" as in:
When published this resource will have a digital object identifier (DOI) http://dx.doi.org/10.4211/hs.877bf9ed9e66468cadddb229838a9ced

**Comment by dtarb** [comment]
Note also that the DOI system now does not need the dx in URL's so we should consider using DOI URL's in the format  http://doi.org/10.4211/hs.877bf9ed9e66468cadddb229838a9ced.  This also applies to the How to cite field.
